barretenberg/ts/src/types/fields.ts

```diff
@@ -10,14 +11,14 @@ export class Fr {
   value: Uint8Array;
 
   constructor(value: Uint8Array | bigint) {
-    if (typeof value === 'bigint') {
-      if (value > Fr.MAX_VALUE) {
-        throw new Error(`Fr out of range ${value}.`);
-      }
-      this.value = toBufferBE(value);
-    } else {
-      this.value = value;
+    // We convert buffer value to bigint to be able to check it fits within modulus
+    const valueBigInt = typeof value === 'bigint' ? value : toBigIntBE(value);
+
+    if (valueBigInt > Fr.MAX_VALUE) {
+      throw new Error(`Fr out of range: ${valueBigInt}`);
     }
+
+    this.value = typeof value === 'bigint' ? toBufferBE(value) : value;
```

Comment:
The code here was not performing modulus check when you passed a buffer on input and we were passing in random 32 bytes in plenty of the places. Not sure why it has not caused issues but I assume it's because we passed that only in a few unit tests (in e2e we always get "real" fields).